superset/assets/src/SqlLab/actions.js

```diff
@@ -405,6 +405,25 @@ export function popSavedQuery(saveQueryId) {
     });
   };
 }
+export function popDatasourceQuery(datasourceKey, sql) {
+  return function (dispatch) {
+    $.ajax({
+      type: 'GET',
+      url: `/superset/fetch_datasource_metadata?datasourceKey=${datasourceKey}`,
+      success: (metadata) => {
+        const queryEditorProps = {
+          title: 'Query ' + metadata.name,
+          dbId: metadata.database.id,
+          schema: metadata.schema,
+          autorun: sql !== undefined,
+          sql: sql || metadata.select_star,
+        };
+        dispatch(addQueryEditor(queryEditorProps));
+      },
+      error: () => notify.error(t('The datasource couldn\'t be loaded')),
```

Comment:
nit (optional): the linter allows double quotes when the string contains a single quote, so that you don't have to escape it